828-Tower-of-Hanoi-Java.txt

```diff
@@ -0,0 +1,43 @@
+
+/**
+ * Tower of Hanoi. Recursive method is used.
+ * Author - Khushee Kapoor
+ * Shift(N,A,B,C)
+ * A,B,C are 3 pegs such that peg A contains N number of discs. The discs are to be moved from peg A to 
+ * peg C in ascending order of their sizes. The method used for this purposive is recursive method.
+ * Algorithm used:
+ * 1. [Set the base case]
+ *      If(N==1)
+ *      Display("Moving Disc A -> C")
+ *      else
+ * 2. [Recurisve Case]
+ *      Shift(N-1,A,C,B)
+ *      Display("Moving Disc A -> C")
+ *      Shift(N-1,B,A,C)
+ */
+import java.util.*;
+public class Tower_of_Hanoi
+{
+   public void Shift(int N, String first, String middle, String last)
+   {
+       if (N==1)
+       { 
+           System.out.println("Disc moved from \t"+first+"->"+last);
+       }
+       else
+       {
+           Shift(N-1, first, last, middle);
+           System.out.println("Disc moved from \t"+first+"->"+last);
+           Shift(N-1, middle, first, last);
+        }
+    }
+    
+   public static void main()
+   {
+       Tower_of_Hanoi ob = new Tower_of_Hanoi();
+       System.out.println("Enter number of discs:");
+       Scanner scanner = new Scanner(System.in);
+       int N = scanner.nextInt();
+       ob.Shift(N,"A","B","C");
+    }
+}
```

Comment:
Add sample input and sample output

Reply:
Done, thank you for your guidance. Really appreciated by a beginner :)